src/test/utilities/collectionUtils.ts

```diff
@@ -10,6 +10,12 @@ export async function* asyncGenerator<T>(items: T[]): AsyncIterableIterator<T> {
     yield* items
 }
 
+export function intoCollection<T>(arr: T[]): AsyncCollection<T> {
+    return toCollection(async function* () {
+        yield* arr
+    })
+}
+
 export function createCollectionFromPages<T>(...pages: T[]): AsyncCollection<T> {
```

Comment:
what is test-specific about these utils? why don't they live in `src/shared/utilities/collectionUtils.ts` ?

Reply:
I think all the methods in here are only currently used in tests. My understanding of these methods is that they allow us to create a `AsyncCollection` from non-async objects.  Would there ever be a use case for this outside of testing?

Reply:
Not sure, but if there is a use-case outside of testing they would be less likely to be discovered. When choosing to create "test libraries" it should be obvious that the library stuff is only relevant to tests. If it's not obvious, why not put it in the common shared library?

Reply:
I setup a separate PR for the refactor which could be merged into here first, before merging it all together. Let me know if I understood what your saying correctly. https://github.com/aws/aws-toolkit-vscode/pull/3535